app_specific_files/dummy_app_multicast/scripts/task0.py

```diff
@@ -0,0 +1,103 @@
+import os
+import time
+import shutil
+import math
+import random
+
+def task(filename,pathin,pathout):
+	execution_time = 3.1
+	timeout = time.time() + execution_time
+	while time.time() < timeout:
+		1+1
+	task_name = os.path.basename(__file__).split('.')[0]
+	print('-------------------------')
+	print(task_name)
+	print(filename)
+	print(pathin)
+	print(pathout)
+	print('-------------------------')
+	if isinstance(filename, list):
+		output1_list = [file.split('.')[0] +'_'+task_name+'.txt' for file in filename]
+		input_file = filename[0].split('_')[0]
+	elif not isinstance(filename, list):
+		output1_list=[filename.split('.')[0] +'_'+task_name+'.txt']
+		input_file = filename.split('_')[0]
+	print(output1_list)
+	output1=set(output1_list)
+	print(output1)
+	print(input_file)
+	output_fname=[f.split('.')[0].split('_')[-1] for f in output1]
+	output_name='_'.join(output_fname)
+	output_name=input_file+'_'+output_name
+	print(output_name)
+	print('-------------------------@@@')
+	print(os.path.realpath('communication.txt'))
+	f = open('/centralized_scheduler/communication.txt', 'r')
+	total_info = f.read().splitlines()
+	f.close()
+	comm = dict()
+	multicast = dict()
+	for line in total_info:
+		src = line.strip().split(' ')[0]
```

Comment:
There is a lot of filename and file processing. This begs the question - should we be creating a parsing library?

Reply:
Yes, this should be done in the future. Actually it is better to have a file handler with any kind of files rather than only current text files.. This is a major change. Currently it depends on the file text name to handle transfering from node to node.

Reply:
We're going to need to think about this otherwise integrating the app by Krishna will be painful.